Source/WebCore/loader/SubresourceIntegrity.cpp

```diff
@@ -161,6 +162,87 @@ static Vector<EncodedResourceCryptographicDigest> strongestMetadataFromSet(Vecto
     return result;
 }
 
+static Ref<FormData> createReportFormData(const String& type, const URL& url, const String& userAgent, const Function<void(JSON::Object&)>& populateBody)
+{
+    auto body = JSON::Object::create();
+    populateBody(body);
+
+    // https://www.w3.org/TR/reporting-1/#queue-report, step 2.3.1.
+    auto reportObject = JSON::Object::create();
+    reportObject->setObject("body"_s, WTFMove(body));
+    reportObject->setString("type"_s, type);
+    reportObject->setString("user_agent"_s, userAgent);
+    reportObject->setInteger("age"_s, 0); // We currently do not delay sending the reports.
+    reportObject->setInteger("attempts"_s, 0);
+    if (url.isValid())
+        reportObject->setString("url"_s, url.strippedForUseAsReferrer().string);
+
+    auto reportList = JSON::Array::create();
+    reportList->pushObject(reportObject);
+
+    return FormData::create(reportList->toJSONString().utf8());
+}
+
+static String addHashPrefix(ResourceCryptographicDigest::Algorithm algorithm, const String& hash)
+{
+    switch (algorithm) {
+    case ResourceCryptographicDigest::Algorithm::SHA256:
+        return makeString("sha256-"_s, hash);
+    case ResourceCryptographicDigest::Algorithm::SHA384:
+        return makeString("sha384-"_s, hash);
+    case ResourceCryptographicDigest::Algorithm::SHA512:
+        return makeString("sha512-"_s, hash);
+    }
+    ASSERT_NOT_REACHED();
+    return String();
+}
+
+static std::optional<ResourceCryptographicDigest::Algorithm> getStrongestAlgorithm(HashAlgorithmSet algorithmSet)
```

Comment:
```suggestion
static std::optional<ResourceCryptographicDigest::Algorithm> findStrongestAlgorithm(HashAlgorithmSet algorithmSet)
```
WebKit coding style does not use _get_ for function names like this, reserving it for functions with "out arguments". Leaving the verb _find_ is OK, though. Or it could just be named `strongestAlgorithm`.

Reply:
done!